src/decide.ts

```diff
@@ -227,84 +267,89 @@ function releaseGroupDoesMatch(
+	}
 
+	const result = await snatch(candidate);
 	if (result.isErr()) {
-		return result.unwrapErr() === SnatchError.RATE_LIMITED
+		const err = result.unwrapErr();
+		return err === SnatchError.RATE_LIMITED
 			? { decision: Decision.RATE_LIMITED }
-			: { decision: Decision.DOWNLOAD_FAILED };
+			: err === SnatchError.MAGNET_LINK
+				? { decision: Decision.MAGNET_LINK }
+				: { decision: Decision.DOWNLOAD_FAILED };
 	}
+	const metafile = result.unwrap();
+	cacheTorrentFile(metafile);
+	candidate.size = metafile.length; // Trackers can be wrong
 
-	const candidateMeta = result.unwrap();
-
-	if (hashesToExclude.includes(candidateMeta.infoHash)) {
-		return { decision: Decision.INFO_HASH_ALREADY_EXISTS };
+	if (hashesToExclude.includes(metafile.infoHash)) {
+		return { decision: Decision.INFO_HASH_ALREADY_EXISTS, metafile };
 	}
-	const sizeMatch = compareFileTreesIgnoringNames(candidateMeta, searchee);
-	const perfectMatch = compareFileTrees(candidateMeta, searchee);
 
+	const perfectMatch = compareFileTrees(metafile, searchee);
 	if (perfectMatch) {
-		return { decision: Decision.MATCH, metafile: candidateMeta };
+		return { decision: Decision.MATCH, metafile };
 	}
-	if (
-		sizeMatch &&
-		matchMode !== MatchMode.SAFE &&
-		searchee.files.length === 1
-	) {
-		return {
-			decision: Decision.MATCH_SIZE_ONLY,
-			metafile: candidateMeta,
-		};
+
+	const sizeMatch = compareFileTreesIgnoringNames(metafile, searchee);
+	if (sizeMatch && matchMode !== MatchMode.SAFE) {
+		return { decision: Decision.MATCH_SIZE_ONLY, metafile };
 	}
 
 	if (matchMode === MatchMode.PARTIAL) {
-		const partialSizeMatch = compareFileTreesPartialIgnoringNames(
-			candidateMeta,
-			searchee,
-		);
+		const partialSizeMatch = comparePartialSizeOnly(metafile, searchee);
 		if (!partialSizeMatch) {
-			return { decision: Decision.SIZE_MISMATCH };
+			return { decision: Decision.PARTIAL_SIZE_MISMATCH, metafile };
 		}
-
-		const partialMatch = compareFileTreesPartial(candidateMeta, searchee);
+		const partialMatch = compareFileTreesPartial(metafile, searchee);
 		if (partialMatch) {
-			return {
-				decision: Decision.MATCH_PARTIAL,
-				metafile: candidateMeta,
-			};
+			return { decision: Decision.MATCH_PARTIAL, metafile };
 		}
 	} else if (!sizeMatch) {
-		return { decision: Decision.SIZE_MISMATCH };
+		return { decision: Decision.SIZE_MISMATCH, metafile };
 	}
 
-	return { decision: Decision.FILE_TREE_MISMATCH };
+	return { decision: Decision.FILE_TREE_MISMATCH, metafile };
 }
 
-function existsInTorrentCache(infoHash: string): boolean {
-	return existsSync(
-		path.join(appDir(), TORRENT_CACHE_FOLDER, `${infoHash}.cached.torrent`),
+function existsInTorrentCache(infoHash: string | undefined): boolean {
+	if (!infoHash) return false;
+	const torrentPath = path.join(
+		appDir(),
+		TORRENT_CACHE_FOLDER,
+		`${infoHash}.cached.torrent`,
 	);
+	if (!existsSync(torrentPath)) return false;
+	utimesSync(torrentPath, new Date(), new Date());
```

Comment:
this changes the modification time even though we're not modifying the file. would it be possible to just change the access time?

Reply:
Yes but from my understanding, modified time is the only thing guaranteed to be stable across operating systems. It seems as we can rely on access time for our use cases though. But some user configurations (I think very unlikely) might not use access time.

This will only matter if we want to clean up cached torrent files. Say older than `Math.max(excludeOlder, ms("1y"))`.

Reply:
what very unlikely user configurations would those be?

Reply:
If the user sets `noatime` for their mount in linux, then it won't be tracked by the filesystem. We can only update atime for now but if we want to use it for clearing cache, modified avoids any potential issues.

Reply:
eh, that seems esoteric enough that I'm inclined not to worry about it. I'd rather have the semantics be right. The alternative would be to track them in the database but that feels heavier than necessary.